build.sbt

```diff
@@ -21,26 +20,38 @@ val jacksons = Seq(
 ).map(_ % jacksonVersion)
 
 def jsonDependencies(scalaVersion: String) = Seq(
-  "org.scala-lang" % "scala-reflect" % scalaVersion,
-  logback % Test
-) ++ jacksons ++ specsBuild.map(_ % Test)
-
+  "org.scala-lang" % "scala-reflect" % scalaVersion
+)
 
 lazy val root = project
   .in(file("."))
-  .enablePlugins(PlayRootProject)
-  .aggregate(`play-json`, `play-functional`)
+  .enablePlugins(PlayRootProject, ScalaJSPlugin)
+  .aggregate(`play-jsonJS`, `play-jsonJVM`,
+    `play-functionalJS`, `play-functionalJVM`)
 
-lazy val `play-json` = project
+lazy val `play-json` = crossProject.crossType(CrossType.Full)
   .in(file("play-json"))
   .enablePlugins(PlayLibrary)
-  .settings(libraryDependencies ++= jsonDependencies(scalaVersion.value))
+  .settings(libraryDependencies ++= jsonDependencies(scalaVersion.value) ++ Seq(
+    "org.scalatest" %%% "scalatest" % "3.0.0" % Test
```

Comment:
Use scalatest instead of Specs2 for the tests about the shared code, as compatible with ScalaJS